pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -233,33 +233,28 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false, isUpdate)
 	case workflowv1alpha1.WorkflowStateTerminated:
-		logCtx.Info("Workflow return state=Terminated")
-		handler.UpdateApplicationRevisionStatus(logCtx, handler.latestAppRev, false, app.Status.Workflow)
-		r.doWorkflowFinish(app, workflowState)
+		if workflowInstance.Status.EndTime.IsZero() {
```

Comment:
we need to remove this endtime when workflow need to re-run

Reply:
Yes, it will. The end time will be cleared when workflow restarts.